### PR TITLE
Issue #7185: Delete thumbnails when a session is removed

### LINKED ISSUE #7185: Delete thumbnails when a session is removed
If we continue to follow the direction of using the session ID for identifying tabs, we should also make sure we remove the thumbnails for the tab when the tab is removed.

We can do this using the same `ThumbnailMiddleware` we added for getting new thumbnails.

### PR DESCRIPTION
**Pull request by gabrielluong**
---

Fixes #7185

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by gabrielluong** [comment]
I think we need to move ahead with simply removing the associated thumbnail whenever the session is removed, and not handle the undo case as a first iteration. We can be filling up the disk cache and provide no way to clear it, which isn't great otherwise.

File: components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/storage/ThumbnailStorage.kt [hunk 1]
```diff
@@ -42,6 +42,24 @@ class ThumbnailStorage(
         context.resources.getDimensionPixelSize(R.dimen.mozac_browser_thumbnails_maximum_size)
     private val scope = CoroutineScope(jobDispatcher)
 
+    /**
+     * Clears all the stored thumbnails in the disk cache.
+     */
+    fun clearThumbnails() {
+        logger.debug("Cleared all thumbnails from disk")
```
Comment (jonalmeida) [inline components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/storage/ThumbnailStorage.kt:49]: This should be a in coroutine scope too.

File: components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/utils/ThumbnailDiskCache.kt [hunk 2]
```diff
@@ -24,7 +23,6 @@ class ThumbnailDiskCache {
     private var thumbnailCache: DiskLruCache? = null
     private val thumbnailCacheWriteLock = Any()
 
-    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     internal fun clear(context: Context) {
         getThumbnailCache(context).delete()
```
Comment (jonalmeida) [inline components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/utils/ThumbnailDiskCache.kt:27]: Must have missed this during the previous review, shouldn't this be in a synchronized block as well?

File: components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/storage/ThumbnailStorage.kt (hunk 1 above)
Comment (gabrielluong) [reply components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/storage/ThumbnailStorage.kt:49]: Fixed.

**Comment by jonalmeida** [comment]
Whoops, approved the wrong PR. We should still fix the other review comment.

File: components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/utils/ThumbnailDiskCache.kt (hunk 2 above)
Comment (gabrielluong) [reply components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/utils/ThumbnailDiskCache.kt:27]: Fixed.

**Comment by gabrielluong** [comment]
bors r=jonalmeida

**Pull request merged**